WordPress/Classes/ViewRelated/Reader/ReaderSubscriptionsView.swift

```diff
@@ -0,0 +1,191 @@
+        .onReceive(subscriptions.publisher) { _ in
+            if !searchText.isEmpty {
+                reloadSearchResults(searchText: searchText)
+            }
+        }
+        .onChange(of: searchText) {
+            reloadSearchResults(searchText: $0)
+        }
+        .toolbar {
+            EditButton()
+        }
+    }
+
+    private func makeSubscriptionCell(for site: ReaderSiteTopic) -> some View {
+        Button {
+            onSelection(site)
+        } label: {
+            ReaderSubscriptionCell(site: site, onDelete: delete)
+        }
+    }
+
+    private var buttonAddSubscription: some View {
+        Button {
+            isAddingSubscription = true
+        } label: {
+            HStack(spacing: 16) {
+                Image(systemName: "plus.circle.fill")
+                    .foregroundStyle(AppColor.brand, Color(.secondarySystemFill))
+                    .font(.largeTitle.weight(.light))
+                    .frame(width: 40)
+                    .padding(.leading, 4)
+                VStack(alignment: .leading) {
+                    Text(Strings.addSubscription)
+                        .font(.callout.weight(.medium))
+                    Text(Strings.addSubscriptionSubtitle)
+                        .font(.footnote)
+                        .foregroundStyle(.secondary)
+                }
+                .lineLimit(1)
+            }
+            .padding(.bottom, 4)
+        }
+        .buttonStyle(.plain)
+        .popover(isPresented: $isAddingSubscription) {
+            ReaderAddSubscriptionView()
+        }
+    }
+
+    private func delete(at offsets: IndexSet) {
+        for site in offsets.map(getSubscription) {
+            delete(site)
+        }
+    }
+
+    private func getSubscription(at index: Int) -> ReaderSiteTopic {
+        if isShowingSearchResuts {
+            searchResults[index]
+        } else {
+            subscriptions[index]
+        }
+    }
+
+    private func delete(_ site: ReaderSiteTopic) {
+        ReaderSubscriptionHelper.unfollow(site)
+    }
+
+    private func reloadSearchResults(searchText: String) {
+        let ranking = StringRankedSearch(searchTerm: searchText)
+        searchResults = ranking.search(in: subscriptions) { "\($0.title) \($0.siteURL)" }
+    }
+}
+
+struct ReaderAddSubscriptionView: View {
+    @State private var url = ""
+    @FocusState private var isFocused: Bool
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 8) {
+            VStack(alignment: .leading) {
+                Text(Strings.addSubscription)
+                    .font(.title3.weight(.medium))
+                Text(Strings.addSubscriptionSubtitle)
+                    .font(.callout)
+                    .foregroundStyle(.secondary)
+            }
+            HStack {
+                TextField("Link", text: $url, prompt: Text(verbatim: "https://www.site.com"))
+                    .focused($isFocused)
```

Comment:
Setting `textContentType`, disabling auto-completion, and maybe other things that help to make typing url eaiser?

Reply:
Done